plugin/evm/limit_order.go

```diff
@@ -152,7 +168,113 @@ func (lop *limitOrderProcesser) listenAndStoreLimitOrderTransactions() {
 }
 
 func (lop *limitOrderProcesser) handleChainAcceptedEvent(event core.ChainEvent) {
+	// it is very important to hold the lock here so that there are no updates to the DB
+	// between Accept() and saveMemoryDBSnapshot()
+	lop.memoryDb.Lock()
```

Comment:
take the lock after 180 if we need to take a snapshot

Reply:
Suppose block 1001 is added as head and its has 5 events.
While those events are being processed in  contracts_events_processor.go , we got Accepted for 1000 block.
Now as per logic if we take a snapshot while the events for Head block -1001 are being processed. We may take a snapshot in between. Later when we revert the logs for 1001 we will be left with bad state.

Reply:
Yep, I was also not so happy with locking memory DB from outside. So I've added locking at Limit order processor level. If those 5 events are being processed then limit order processor will be locked and only after that will the snapshot proceed